src/proctored-exam-settings/ProctoredExamSettings.test.jsx

```diff
@@ -113,8 +113,20 @@ describe('ProctoredExamSettings tests with escalation email', () => {
     await act(async () => {
       fireEvent.click(selectButton);
     });
+
+    // verify alert content and focus management
     const escalationEmailError = screen.getByTestId('proctortrackEscalationEmailError');
     expect(escalationEmailError.textContent).not.toBeNull();
+    expect(document.activeElement).toEqual(escalationEmailError);
+
+    // verify alert link links to offending input
+    const errorLink = screen.getByTestId('proctorTrackEscalationEmailErrorLink');
+    await act(async () => {
+      fireEvent.click(errorLink);
```

Comment:
Clicking the link doesn't seem to do move focus in the test. The focus remains on the Alert component and does not move to the input. I have put in a debugger, and it does seem to run the useEffect hook that handles focus management, so I'm not sure what's going on.

I keep getting this error in the console.

`TypeError: 'set' on proxy: trap returned falsish for property 'Symbol(impl)'`

Reply:
Please read https://testing-library.com/docs/guide-events. Looks like you should call `element.focus()` to move your focus.

Reply:
I'm not trying to test what happens when I call focus on an element, which would be valuable if I had an `onFocus` handler I wanted to test. I'm trying to test that clicking on a link moves the focus to the linked element. I don't want the test to call the focus method on the input, because I'm expecting that clicking the link will do that for me.

Reply:
@MichaelRoytman Ah, got it. Then according to the doc I linked, firing such a `click` event won't get you the desired testing behavior.
What needs to happen if you really want to test this, is to explicitly have an event handler in this element for `onclick`, then in their, update the focus of the element. Making this auto focus functionality a javascript event thing.

Reply:
Ah, I see. I think that I was confused because Chrome sets the focus automatically for you. But Firefox, for example, doesn't. I'll add explicit focus management to make sure it works in all browsers.